pkg/nuctl/command/import.go

```diff
@@ -0,0 +1,375 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/platform"
+
+	"github.com/ghodss/yaml"
+	"github.com/nuclio/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+type importCommandeer struct {
+	cmd            *cobra.Command
+	rootCommandeer *RootCommandeer
+	deploy         bool
+}
+
+func newImportCommandeer(rootCommandeer *RootCommandeer) *importCommandeer {
+	commandeer := &importCommandeer{
+		rootCommandeer: rootCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import function/project",
+	}
+
+	importFunctionCommand := newImportFunctionCommandeer(commandeer).cmd
+	importProjectCommand := newImportProjectCommandeer(commandeer).cmd
+
+	cmd.AddCommand(
+		importFunctionCommand,
+		importProjectCommand,
+	)
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (i *importCommandeer) addImportCommandFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&i.deploy, "deploy", false, "Deploy the function or functions after import, false by default")
+}
+
+func (i *importCommandeer) readFromStdinOrFile(args []string) ([]byte, error) {
+	if len(args) >= 1 {
+		return ioutil.ReadFile(args[0])
+	}
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func (i *importCommandeer) getUnmarshalFunc(bytes []byte) (func(data []byte, v interface{}) error, error) {
+	var err error
+	var obj map[string]interface{}
+
+	if err = json.Unmarshal(bytes, &obj); err == nil {
+		return json.Unmarshal, nil
+	}
+
+	if err = yaml.Unmarshal(bytes, &obj); err == nil {
+		return yaml.Unmarshal, nil
+	}
+
+	return nil, errors.New("Input is neither json nor yaml")
+}
+
+func (i *importCommandeer) importFunction(functionConfig *functionconfig.Config, deploy bool, project string) error {
```

Comment:
Why is this a thing?
Like we only have `POST /functions` to import an exported function, `nuctl deploy function` should take care of import

Reply:
Mainly for the import project logic which does ensure things (like project name) in the dashboard as well.
And for singular functions, in ductal, the difference is - the `--deploy` flag which includes the ability to deploy on import.